src/app/api/[[...route]]/routes/tenant.route.ts

```diff
@@ -14,12 +19,89 @@ app.delete('/:tenantId', (c) => {
   return c.json({ message: `Deleting tenant id: ${tenantId}` });
 });
 
-app.post('/:shareHouseId', (c) => {
-  const shareHouseId = c.req.param('shareHouseId');
-  return c.json({
-    message: `Creating share house id for tenant: ${shareHouseId}`,
-  });
-});
+app.post(
+  '/:shareHouseId',
+  zValidator('json', tenantInvitationSchema),
+  async (c) => {
+    try {
+      const shareHouseId = c.req.param('shareHouseId');
+      const data = c.req.valid('json');
+
+      const existingTenant = await prisma.tenant.findUnique({
+        where: {
+          email: data.email,
+        },
+      });
+
+      if (existingTenant)
+        return c.json({ error: 'Tenant with this email already exists' }, 400);
```

Comment:
Could you sanitize the email address by lowercasing it and removing unnecessary spaces before searching for the tenant?

Reply:
I added a feature that sanitizes email addresses.